CRM/Contribute/Form/ContributionPage/Amount.php

```diff
@@ -337,6 +335,7 @@ public static function formRule($fields, $files, $self) {
     if (isset($fields['is_pay_later'])) {
       if (empty($fields['pay_later_text'])) {
         $errors['pay_later_text'] = ts('Please enter the text for the \'pay later\' checkbox displayed on the contribution form.');
+        $this->addElement('checkbox', 'is_monetary', ts('Execute real-time monetary transactions'));
```

Comment:
So, I believe the is_monetary (execute real time transactions) field has been redundant for as long as I can remember - ie. we can calculate whether there is an amount and we can calculate whether 'is_pay_later' is set without this field. It seems to introduce a potential for error with no gain.

In terms of this field the manual processor concept is tangental - ie. we don't need the field with, or without that concept being in play.

Re $0 transactions - I think you are probably right.